app/oxalis/security/CombinedAuthenticatorService.scala

```diff
@@ -0,0 +1,472 @@
+package oxalis.security
+
+import com.mohiva.play.silhouette.api._
+import com.mohiva.play.silhouette.api.exceptions._
+import com.mohiva.play.silhouette.api.services.{AuthenticatorResult, AuthenticatorService}
+import com.mohiva.play.silhouette.api.util.{Base64, Clock, FingerprintGenerator, IDGenerator}
+import com.mohiva.play.silhouette.impl.authenticators.CookieAuthenticatorService.{InvalidJsonFormat, JsonParseError}
+import com.mohiva.play.silhouette.impl.authenticators._
+import com.mohiva.play.silhouette.impl.util.CookieSigner
+import play.api.libs.Crypto
+import play.api.libs.json.Json
+import com.mohiva.play.silhouette.impl.daos.AuthenticatorDAO
+import org.joda.time.DateTime
+import play.api.http.HeaderNames
+import play.api.libs.Crypto
+import play.api.libs.json.Json
+import play.api.mvc._
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration.FiniteDuration
+import scala.util.{Failure, Success, Try}
+import com.mohiva.play.silhouette._
+import play.api.http.HeaderNames
+import CombinedAuthenticatorService._
+import scala.concurrent.duration._
+import scala.concurrent.{ ExecutionContext, Future }
+import play.api.libs.json._
+import play.api.libs.json.Reads._
+import play.api.libs.json.Writes._
+/**
+  * Created by robert on 10.11.17.
+  *
+  * The code is adjusted to fit with our own CombinedAuthenticator.
+  * The original code is from silhouette.impl.authenticators.CookieAuthenticator
+  */
```

Comment:
could you make this comment longer, briefly describing what exactly you changed with respect to the original `silhouette.impl.authenticators.CookieAuthenticator`?

Reply:
we might also have to add a copy of their apache license… is there no way we can do this without copying their code?